Assets/Scripts/Models/FurnitureWorkshop.cs

```diff
@@ -0,0 +1,343 @@
+        //   - as output will be produced after time, it is possible that output spot can be ocupied meanwhile
+        // - process for specified time
+        // - if output slot is free, provide output (if not, keep output 'inside' factory)
+
+        var curSetupChainName = furnParameters[CUR_PRODUCTION_CHAIN_PARAM_NAME].ToString();
+
+        if (!string.IsNullOrEmpty(curSetupChainName))
+        {
+            FurnitureWorkshop.ProductionChain prodChain = GetProductionChainInfo(curSetupChainName);
+            // if there is no processing in progress
+            if (furnParameters[CUR_PROCESSED_INV_PARAM_NAME].ToInt() == 0)
+            {
+                // check input slots for input inventory               
+                List<KeyValuePair<Tile, int>> flaggedForTaking = CheckForInventoryAtInput(prodChain);
+                // if all the input requirements are ok, you can start processing:
+                if (flaggedForTaking.Count == prodChain.Input.Count)
+                {
+                    // consume input inventory
+                    ConsumeInventory(flaggedForTaking);
+
+                    furnParameters[CUR_PROCESSED_INV_PARAM_NAME].SetValue(prodChain.Output.Count);
+
+                    // reset processing timer and set max time for processing for this prod. chain
+                    furnParameters[CUR_PROCESSING_TIME_PARAM_NAME].SetValue(0f);
+                    furnParameters[MAX_PROCESSING_TIME_PARAM_NAME].SetValue(prodChain.ProcessingTime);
+                }
+            }
+            else
+            {
+                // processing is in progress
+                furnParameters[CUR_PROCESSING_TIME_PARAM_NAME].ChangeFloatValue(deltaTime);
+
+                if (furnParameters[CUR_PROCESSING_TIME_PARAM_NAME].ToFloat() >=
+                    furnParameters[MAX_PROCESSING_TIME_PARAM_NAME].ToFloat())
+                {
+                    List<TileObjectTypeAmount> outPlacement = CheckForInventoryAtOutput(prodChain);
+
+                    // if output placement was found for all products, place them
+                    if (outPlacement.Count == prodChain.Output.Count)
+                    {
+                        PlaceInventory(outPlacement);
+                        // processing done, can fetch input for another processing
+                        furnParameters[CUR_PROCESSED_INV_PARAM_NAME].SetValue(0);
+                    }
+                }
+            }
+            // create possible jobs for factory(hauling input)
+            HaulingJobForInputs(prodChain);
+        }
+    }
+
+    private void HaulingJobForInputs(FurnitureWorkshop.ProductionChain prodChain)
+    {
+        // for all inputs in production chain
+        foreach (var reqInputItem in prodChain.Input)
+        {
+            // if there is no hauling job for input object type, create one
+            var existingHaulingJob = furniture.jobs.Any(x => x.inventoryRequirements.ContainsKey(reqInputItem.ObjectType));
```

Comment:
is there a reason to keep it private there? exposing collection enables linq queries for various needs (even for modders?). People will be adding their queries to furniture all the time.. is that desired?
is it that with getter on collections you are actually able to add items to it?
there is another PR #1201 that exposes only indexer to jobs